src/ripple/protocol/impl/SecretKey.cpp

```diff
@@ -178,16 +147,27 @@ randomSecretKey()
 
 // VFALCO TODO Rewrite all this without using OpenSSL
 //             or calling into GenerateDetermisticKey
-
 SecretKey
-generateSecretKey (Seed const& seed)
+generateSecretKey (KeyType type, Seed const& seed)
 {
-    uint128 ps;
-    std::memcpy(ps.data(),
-        seed.data(), seed.size());
-    auto const upk =
-        generateRootDeterministicPrivateKey(ps);
-    return SecretKey(Slice{ upk.data(), upk.size() });
+    if (type == KeyType::ed25519)
+    {
+        auto const key = sha512Half_s(Slice(
+            seed.data(), seed.size()));
+        return SecretKey(Slice{ key.data(), key.size() });
+    }
+
+    if (type == KeyType::secp256k1)
+    {
+        uint128 ps;
+        std::memcpy(ps.data(),
+            seed.data(), seed.size());
+        auto const upk =
+            generateRootDeterministicPrivateKey(ps);
+        return SecretKey(Slice{ upk.data(), upk.size() });
+    }
```

Comment:
When it comes to secp256k1 keys, we generate "node" and "account" keys differently. Ideally, we'll update this code so that we have a `generateNodeKeypair` and `generateAccountKeyPair` which take a `Seed` and internally do the right thing.